### PR TITLE
Some README improvements

### PR DESCRIPTION
**Pull request by hugomg**
Based on my adventures trying to compile RIR on Fedora (bad idea)

1. Strongly recommend Ubuntu, due to fetch-llvm
2. Tell how to install the compilation dependencies
3. Encourage off-tree builds and ninja by default
4. Gather the macOS instructions in a single place

[skip ci]

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,44 +1,86 @@
 # Ř
 
-To give it a spin try
+## Giving it as spin
```
Comment (o-) [inline README.md]: *a* spin

File: README.md [hunk 2]
```diff
@@ -1,44 +1,86 @@
...
+    # The cmake should output "Found LLVM 12"
+    cmake ..
+    ninja
+
+Congratulations! You can now run Ř with
 
     bin/R
 
-If you want to develop Ř then it is highly recommended to use off-tree builds and ninja instead of make. See further down (Hacking).
+### In-tree builds
```
Comment (o-) [inline README.md]: drop? is there any need for that?

**Review by o- — COMMENTED** [review]
good stuff,  thanks

File: README.md (hunk 2 above)
Comment (hugomg) [reply README.md]: Sounds reasonable, YAGNI. I suppose that if someone really wants in-tree builds, they are free to figure out the cmake commands themselves.

**Pull request merged**